lib/reek/source/source_locator.rb

```diff
@@ -8,64 +8,27 @@ module Source
     # Finds Ruby source files in a filesystem.
     #
     class SourceLocator
+      include Enumerable
+      extend Forwardable
+
+      def_delegator :sources, :each
       # Initialize with the paths we want to search.
       #
       # paths - a list of paths as Strings
       def initialize(paths, configuration: Configuration::AppConfiguration.default)
-        @paths = paths.flat_map do |string|
-          path = Pathname.new(string)
-          current_directory?(path) ? path.entries : path
+        @paths = paths.map do |string|
+          SourcePath.new(Pathname.new(string), configuration: configuration)
```

Comment:
How about moving the 

```Ruby
Pathname.new(string)
```

into SourcePath's initializer to hide away this complexity?